### ISSUE TITLE
`TestOp` init broken?

### ISSUE THREAD

**Issue by AntonLydike** [xdslproject/xdsl#1143]
Construction a testop using it's init doesn't seem to work?

```
test.TestOp(result_types=[builtin.i32, builtin.f32])
```
produces:
```
E           ValueError: Expected 1 operand, but got 0
```

But using `test.TestOp.create(result_types=[builtin.i32, builtin.f32])` works fine...

**Comment by math-fehr** [comment]
This is because `TestOp` has a variadic operand, so it expects one operand in the IRDL form (`operands=[[]]`).
Maybe what you want is that by default, if you have no operands/results/regions, it tries to make everything empty?

**Comment by AntonLydike** [comment]
The __inti__ tries to construct args, but I think it fails?

**Comment by math-fehr** [comment]
Yeah, maybe we should improve the error messages as well.
It is not expecting an operand, but an operand definition, if that makes sense.

**Comment by compor** [comment]
Resolved by #1534 

Bumped into this while playing with the `test` dialect for register allocation stuff
